--- a/addons/allegrogl/src/x.c	2009-10-17 09:05:52.000000000 +0200
+++ a/addons/allegrogl/src/x.c	2018-12-06 21:38:18.768757200 +0100
@@ -650,7 +650,7 @@
 		return -1;
 	}
 
-	if (!(render_type & GLX_RGBA_BIT) && !(render_type & GLX_RGBA_FLOAT_BIT)) {
+	if (!(render_type & GLX_RGBA_BIT) && !(render_type & GLX_RGBA_FLOAT_BIT_ARB)) {
 		TRACE(PREFIX_I "decode_fbconfig: Not RGBA mode\n");
 		return -1;
 	}
@@ -674,7 +674,7 @@
 	/* Floating-point depth is not supported as glx extension (yet). */
 	i->float_depth = 0;
 
-	i->float_color = (render_type & GLX_RGBA_FLOAT_BIT);
+	i->float_color = (render_type & GLX_RGBA_FLOAT_BIT_ARB);
 
 	v = glXGetVisualFromFBConfig(_xwin.display, fbc);
 	if (!v) {
